[GLib] Rename WebKitWebExtension to WebKitWebProcessExtension

mcatanzaro opened the pull request:
#### e60e9083903f323d36875e2195cae5d734e0b692
<pre>
[GLib] Rename WebKitWebExtension to WebKitWebProcessExtension
<a href="https://bugs.webkit.org/show_bug.cgi?id=226665">https://bugs.webkit.org/show_bug.cgi?id=226665</a>

Reviewed by Adrian Perez de Castro.

This should avoid confusion between web process extensions vs.
WebExtensions, the popular and entirely unrelated API for Chrome,
Firefox, and Safari extensions.

* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/PlatformGTKDeprecated.cmake:
* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/PlatformWPEDeprecated.cmake:
* Source/WebKit/Scripts/glib/generate-api-header.py:
* Source/WebKit/Shared/API/glib/WebKitUserMessage.cpp:
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/SourcesWPE.txt:
* Source/WebKit/UIProcess/API/glib/WebKitInjectedBundleClient.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitWebContext.cpp:
(webkit_web_context_class_init):
(createDefaultWebContext): Deleted.
(webkit_web_context_get_default): Deleted.
(webkit_web_context_new): Deleted.
(webkit_web_context_new_ephemeral): Deleted.
(webkit_web_context_new_with_website_data_manager): Deleted.
(webkit_web_context_get_website_data_manager): Deleted.
(webkit_web_context_is_ephemeral): Deleted.
(webkit_web_context_is_automation_allowed): Deleted.
(webkit_web_context_set_automation_allowed): Deleted.
(webkit_web_context_get_network_session_for_automation): Deleted.
(webkit_web_context_set_cache_model): Deleted.
(webkit_web_context_get_cache_model): Deleted.
(webkit_web_context_clear_cache): Deleted.
(webkit_web_context_set_network_proxy_settings): Deleted.
(webkit_web_context_download_uri): Deleted.
(webkit_web_context_get_cookie_manager): Deleted.
(webkit_web_context_get_geolocation_manager): Deleted.
(ensureFaviconDatabase): Deleted.
(webkit_web_context_set_favicon_database_directory): Deleted.
(webkit_web_context_get_favicon_database_directory): Deleted.
(webkit_web_context_get_favicon_database): Deleted.
(webkit_web_context_get_security_manager): Deleted.
(webkit_web_context_set_additional_plugins_directory): Deleted.
(webkit_web_context_get_plugins): Deleted.
(webkit_web_context_get_plugins_finish): Deleted.
(webkit_web_context_register_uri_scheme): Deleted.
(webkit_web_context_set_sandbox_enabled): Deleted.
(pathIsHomeDirectory): Deleted.
(pathIsBlocked): Deleted.
(webkit_web_context_add_path_to_sandbox): Deleted.
(webkit_web_context_get_sandbox_enabled): Deleted.
(webkit_web_context_get_spell_checking_enabled): Deleted.
(webkit_web_context_set_spell_checking_enabled): Deleted.
(webkit_web_context_get_spell_checking_languages): Deleted.
(webkit_web_context_set_spell_checking_languages): Deleted.
(webkit_web_context_set_preferred_languages): Deleted.
(webkit_web_context_set_tls_errors_policy): Deleted.
(webkit_web_context_get_tls_errors_policy): Deleted.
(webkit_web_context_set_web_extensions_directory): Deleted.
(webkit_web_context_set_web_extensions_initialization_user_data): Deleted.
(webkit_web_context_set_disk_cache_directory): Deleted.
(webkit_web_context_prefetch_dns): Deleted.
(webkit_web_context_allow_tls_certificate_for_host): Deleted.
(webkit_web_context_set_process_model): Deleted.
(webkit_web_context_get_process_model): Deleted.
(webkit_web_context_set_web_process_count_limit): Deleted.
(webkit_web_context_get_web_process_count_limit): Deleted.
(addOriginToMap): Deleted.
(webkit_web_context_initialize_notification_permissions): Deleted.
(webkit_web_context_send_message_to_all_extensions): Deleted.
(webkit_web_context_set_use_system_appearance_for_scrollbars): Deleted.
(webkit_web_context_get_use_system_appearance_for_scrollbars): Deleted.
(webkit_web_context_get_time_zone_override): Deleted.
(webkitWebContextInitializeNotificationPermissions): Deleted.
(webkitWebContextDownloadStarted): Deleted.
(webkitWebContextInitializeWebExtensions): Deleted.
(webkitWebContextGetProcessPool): Deleted.
(webkitWebContextCreatePageForWebView): Deleted.
(webkitWebContextWebViewDestroyed): Deleted.
(webkitWebContextGetWebViewForPage): Deleted.
* Source/WebKit/UIProcess/API/glib/WebKitWebContext.h.in:
* Source/WebKit/UIProcess/API/glib/WebKitWebContextPrivate.h:
* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
* Source/WebKit/UIProcess/API/glib/webkit.h.in:
* Source/WebKit/UIProcess/API/gtk/WebKitVersion.h.in:
* Source/WebKit/UIProcess/API/wpe/WebKitVersion.h.in:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitInjectedBundleMain.cpp:
(WKBundleInitialize):
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebExtension.cpp:
(webkitWebProcessExtensionCreate):
(webkitWebProcessExtensionDidReceiveUserMessage):
(webkitWebProcessExtensionSetGarbageCollectOnPageDestroy):
(webkitWebExtensionCreate): Deleted.
(webkitWebExtensionDidReceiveUserMessage): Deleted.
(webkitWebExtensionSetGarbageCollectOnPageDestroy): Deleted.
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebExtension.h.in:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebProcessExtension.cpp: Copied from Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebExtension.cpp.
(webkitWebProcessExtensionCreate):
(webkitWebProcessExtensionDidReceiveUserMessage):
(webkitWebProcessExtensionSetGarbageCollectOnPageDestroy):
(webkit_web_process_extension_get_page):
(webkit_web_process_extension_send_message_to_context):
(webkit_web_process_extension_send_message_to_context_finish):
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebProcessExtension.h.in: Added.
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebProcessExtensionPrivate.h: Renamed from Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebExtensionPrivate.h.
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebProcessExtensionManager.cpp: Renamed from Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitExtensionManager.cpp.
(WebKit::WebProcessExtensionManager::singleton):
(WebKit::WebProcessExtensionManager::scanModules):
(WebKit::parseUserData):
(WebKit::WebProcessExtensionManager::initializeWebProcessExtension):
(WebKit::WebProcessExtensionManager::initialize):
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebProcessExtensionManager.h: Renamed from Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitExtensionManager.h.
(WebKit::WebProcessExtensionManager::extension const):
* Source/WebKit/WebProcess/InjectedBundle/API/glib/webkit-web-process-extension.h.in: Renamed from Source/WebKit/WebProcess/InjectedBundle/API/glib/webkit-web-extension.h.in.
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/WebKitWebProcessEnumTypes.cpp.in:
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/WebKitWebProcessEnumTypesGtk3.h.in:
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/WebKitWebProcessEnumTypesGtk4.h.in:
* Source/WebKit/WebProcess/InjectedBundle/API/wpe/WebKitWebProcessEnumTypes.cpp.in:
* Source/WebKit/WebProcess/InjectedBundle/API/wpe/WebKitWebProcessEnumTypesWPE1.h.in: Copied from Source/WebKit/WebProcess/InjectedBundle/API/wpe/WebKitWebProcessEnumTypes.h.in.
* Source/WebKit/WebProcess/InjectedBundle/API/wpe/WebKitWebProcessEnumTypesWPE2.h.in: Renamed from Source/WebKit/WebProcess/InjectedBundle/API/wpe/WebKitWebProcessEnumTypes.h.in.
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
* Source/WebKit/WebProcess/WebPage/glib/WebPageGLib.cpp:
(WebKit::WebPage::sendMessageToWebProcessExtensionWithReply):
(WebKit::WebPage::sendMessageToWebProcessExtension):
(WebKit::WebPage::sendMessageToWebExtensionWithReply): Deleted.
(WebKit::WebPage::sendMessageToWebExtension): Deleted.
* Source/WebKit/WebProcess/WebProcess.h:
* Source/WebKit/WebProcess/WebProcess.messages.in:
* Source/WebKit/WebProcess/glib/WebProcessGLib.cpp:
(WebKit::WebProcess::sendMessageToWebProcessExtension):
(WebKit::WebProcess::sendMessageToWebExtension): Deleted.
* Source/WebKit/gtk/gtk3-urlmap.js.in: Copied from Source/WebKit/gtk/urlmap.js.in.
* Source/WebKit/gtk/gtk3-webkitgtk.toml.in: Copied from Source/WebKit/gtk/webkitgtk.toml.in.
* Source/WebKit/gtk/gtk4-urlmap.js.in: Renamed from Source/WebKit/gtk/urlmap.js.in.
* Source/WebKit/gtk/gtk4-webkitgtk.toml.in: Renamed from Source/WebKit/gtk/webkitgtk.toml.in.
* Source/WebKit/gtk/migrating-to-webkitgtk-6.0.md:
* Source/WebKit/gtk/webkitgtk-web-process-extension.pc.in: Renamed from Source/WebKit/gtk/webkitgtk-web-extension.pc.in.
* Source/WebKit/gtk/webkitgtk-web-process-extension.toml.in: Renamed from Source/WebKit/gtk/webkitgtk-webextension.toml.in.
* Source/WebKit/webkitglib-symbols.map:
* Source/WebKit/wpe/wpe-web-process-extension-uninstalled.pc.in: Renamed from Source/WebKit/wpe/wpe-web-extension-uninstalled.pc.in.
* Source/WebKit/wpe/wpe-web-process-extension.pc.in: Renamed from Source/WebKit/wpe/wpe-web-extension.pc.in.
* Source/WebKit/wpe/wpe-web-process-extension.toml.in: Renamed from Source/WebKit/wpe/wpewebextension.toml.in.
* Source/WebKit/wpe/wpe1-urlmap.js: Copied from Source/WebKit/wpe/urlmap.js.
(const.baseURLApiLevelSuffix):
* Source/WebKit/wpe/wpe2-urlmap.js: Renamed from Source/WebKit/wpe/urlmap.js.
(const.baseURLApiLevelSuffix):
* Source/WebKit/wpe/wpewebkit.toml.in:
* Source/cmake/OptionsGTK.cmake:
* Source/cmake/OptionsWPE.cmake:
* Tools/PlatformWPE.cmake:
* Tools/Scripts/webkitpy/style/checker.py:
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestLoaderClient.cpp:
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestMultiprocess.cpp:
(testProcessPerWebView):
(testMultiprocessWebViewCreateReadyClose):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebProcessExtensions.cpp: Renamed from Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebExtensions.cpp.
(checkTitle):
(testWebProcessExtensionGetTitle):
(inputElementIsUserEdited):
(testWebProcessExtensionInputElementIsUserEdited):
(documentLoadedCallback):
(testDocumentLoadedSignal):
(webProcessTerminatedCallback):
(testWebKitWebViewProcessCrashed):
(scriptDialogCallback):
(runJavaScriptInIsolatedWorldFinishedCallback):
(testWebProcessExtensionIsolatedWorld):
(didAssociateFormControlsCallback):
(testWebProcessExtensionFormControlsAssociated):
(FormSubmissionTest::FormSubmissionTest):
(FormSubmissionTest::~FormSubmissionTest):
(FormSubmissionTest::testFormSubmissionResult):
(FormSubmissionTest::willSendDOMEventCallback):
(FormSubmissionTest::willCompleteCallback):
(FormSubmissionTest::runJavaScriptAndWaitUntilFormSubmitted):
(testWebProcessExtensionFormSubmissionSteps):
(webViewPageIDChanged):
(testWebProcessExtensionPageID):
(UserMessageTest::webViewUserMessageReceivedCallback):
(UserMessageTest::webContextUserMessageReceivedCallback):
(UserMessageTest::UserMessageTest):
(UserMessageTest::~UserMessageTest):
(UserMessageTest::sendMessage):
(UserMessageTest::sendMessageToAllExtensions):
(UserMessageTest::viewUserMessageReceived):
(UserMessageTest::contextUserMessageReceived):
(UserMessageTest::waitUntilViewMessagesReceived):
(UserMessageTest::waitUntilViewMessageReceived):
(UserMessageTest::waitUntilContextMessageReceived):
(readFileDescriptor):
(testWebProcessExtensionUserMessages):
(testWebProcessExtensionWindowObjectCleared):
(beforeAll):
(afterAll):
* Tools/TestWebKitAPI/Tests/WebKitGLib/WebProcessExtensionTest.cpp: Renamed from Tools/TestWebKitAPI/Tests/WebKitGLib/WebExtensionTest.cpp.
(DelayedSignal::DelayedSignal):
(emitDocumentLoaded):
(documentLoadedCallback):
(emitURIChanged):
(uriChangedCallback):
(sendRequestCallback):
(serializeContextMenu):
(serializeNode):
(contextMenuCallback):
(consoleMessageSentCallback):
(emitFormControlsAssociated):
(formControlsAssociatedForFrameCallback):
(formControlsAssociatedCallback):
(emitFormSubmissionEvent):
(willSubmitFormDeprecatedCallback):
(handleFormSubmissionCallback):
(willSendSubmitEventCallback):
(willSubmitFormCallback):
(pageMessageReceivedCallback):
(emitPageCreated):
(pageCreatedCallback):
(extensionMessageReceivedCallback):
* Tools/TestWebKitAPI/Tests/WebKitGLib/WebProcessTest.cpp:
(windowObjectClearedCallback):
(webkit_web_extension_initialize):
* Tools/TestWebKitAPI/Tests/WebKitGLib/WebProcessTest.h:
* Tools/TestWebKitAPI/Tests/WebKitGtk/AutocleanupsTest.cpp:
* Tools/TestWebKitAPI/Tests/WebKitGtk/DOMClientRectTest.cpp:
* Tools/TestWebKitAPI/Tests/WebKitGtk/DOMDOMWindowTest.cpp:
(WebKitDOMDOMWindowTest::testSignals):
(WebKitDOMDOMWindowTest::testDispatchEvent):
(WebKitDOMDOMWindowTest::testGetComputedStyle):
(WebKitDOMDOMWindowTest::runTest):
* Tools/TestWebKitAPI/Tests/WebKitGtk/DOMNodeFilterTest.cpp:
* Tools/TestWebKitAPI/Tests/WebKitGtk/DOMNodeTest.cpp:
* Tools/TestWebKitAPI/Tests/WebKitGtk/DOMXPathNSResolverTest.cpp:
* Tools/TestWebKitAPI/Tests/WebKitGtk/TestDOMDOMWindow.cpp:
(beforeAll):
* Tools/TestWebKitAPI/glib/CMakeLists.txt:
* Tools/TestWebKitAPI/glib/PlatformGTK.cmake:
* Tools/TestWebKitAPI/glib/TestExpectations.json:
* Tools/TestWebKitAPI/glib/WebKitGLib/TestMain.cpp:
(dbusServerConnection):
* Tools/TestWebKitAPI/glib/WebKitGLib/TestMain.h:
(Test::initializeWebProcessExtensionsCallback):
(Test::Test):
(Test::initializeWebProcessExtensions):
(Test::initializeWebExtensionsCallback): Deleted.
(Test::initializeWebExtensions): Deleted.
* Tools/TestWebKitAPI/glib/WebKitGLib/WebViewTest.cpp:
(WebViewTest::extensionProxy):
* Tools/TestWebKitAPI/glib/WebKitGLib/WebViewTest.h:

Canonical link: <a href="https://commits.webkit.org/261349@main">https://commits.webkit.org/261349@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/aae1bfc3df519a0e6c99a259189a3251ae510500

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/111407 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/20544 "Built successfully") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/90/builds/19 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/87/builds/3214 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/120193 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/115360 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/21912 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/11639 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/87/builds/3214 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/117170 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/16262 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/90/builds/19 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/104126 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/114834 "Passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/98218 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/90/builds/19 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/44935 "layout-tests running") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/1/builds/99938 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/13042 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/90/builds/19 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/86730 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/83/builds/11156 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/13549 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/9435 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/101128 "Built successfully") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/18993 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/90/builds/19 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/31498 "Passed tests") | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7868 "'git push ...'") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/15514 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/109166 "Built successfully") | 
| | | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/26899 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

carlosgcampos commented on Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebProcessExtension.cpp:
```diff
@@ -0,0 +1,312 @@
+     * are always broadcasted to all web extensions and they can't be
+     * replied to. Calling webkit_user_message_send_reply() will do nothing.
+     *
+     * Since: 2.28
```
All Since tags should be 2.40

carlosgcampos commented on Source/WebKit/WebProcess/InjectedBundle/API/glib/WebProcessExtensionManager.h:
```diff
@@ -35,25 +40,25 @@ namespace WebKit {
 class InjectedBundle;
 
-class WebKitExtensionManager {
-    WTF_MAKE_NONCOPYABLE(WebKitExtensionManager);
+class WebProcessExtensionManager {
+    WTF_MAKE_NONCOPYABLE(WebProcessExtensionManager);
 public:
-    __attribute__((visibility("default"))) static WebKitExtensionManager& singleton();
+    __attribute__((visibility("default"))) static WebProcessExtensionManager& singleton();
```
This breaks webkitglib-symbols.map

aperezdc reviewed (approved):
The failure in the `gtk-wk2` builder seems completely unrelated.

mcatanzaro committed:
[GLib] Rename WebKitWebExtension to WebKitWebProcessExtension
https://bugs.webkit.org/show_bug.cgi?id=226665

Reviewed by Adrian Perez de Castro.

This should avoid confusion between web process extensions vs.
WebExtensions, the popular and entirely unrelated API for Chrome,
Firefox, and Safari extensions.

* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/PlatformGTKDeprecated.cmake:
* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/PlatformWPEDeprecated.cmake:
* Source/WebKit/Scripts/glib/generate-api-header.py:
* Source/WebKit/Shared/API/glib/WebKitUserMessage.cpp:
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/SourcesWPE.txt:
* Source/WebKit/UIProcess/API/glib/WebKitInjectedBundleClient.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitWebContext.cpp:
(webkit_web_context_class_init):
(createDefaultWebContext): Deleted.
(webkit_web_context_get_default): Deleted.
(webkit_web_context_new): Deleted.
(webkit_web_context_new_ephemeral): Deleted.
(webkit_web_context_new_with_website_data_manager): Deleted.
(webkit_web_context_get_website_data_manager): Deleted.
(webkit_web_context_is_ephemeral): Deleted.
(webkit_web_context_is_automation_allowed): Deleted.
(webkit_web_context_set_automation_allowed): Deleted.
(webkit_web_context_get_network_session_for_automation): Deleted.
(webkit_web_context_set_cache_model): Deleted.
(webkit_web_context_get_cache_model): Deleted.
(webkit_web_context_clear_cache): Deleted.
(webkit_web_context_set_network_proxy_settings): Deleted.
(webkit_web_context_download_uri): Deleted.
(webkit_web_context_get_cookie_manager): Deleted.
(webkit_web_context_get_geolocation_manager): Deleted.
(ensureFaviconDatabase): Deleted.
(webkit_web_context_set_favicon_database_directory): Deleted.
(webkit_web_context_get_favicon_database_directory): Deleted.
(webkit_web_context_get_favicon_database): Deleted.
(webkit_web_context_get_security_manager): Deleted.
(webkit_web_context_set_additional_plugins_directory): Deleted.
(webkit_web_context_get_plugins): Deleted.
(webkit_web_context_get_plugins_finish): Deleted.
(webkit_web_context_register_uri_scheme): Deleted.
(webkit_web_context_set_sandbox_enabled): Deleted.
(pathIsHomeDirectory): Deleted.
(pathIsBlocked): Deleted.
(webkit_web_context_add_path_to_sandbox): Deleted.
(webkit_web_context_get_sandbox_enabled): Deleted.
(webkit_web_context_get_spell_checking_enabled): Deleted.
(webkit_web_context_set_spell_checking_enabled): Deleted.
(webkit_web_context_get_spell_checking_languages): Deleted.
(webkit_web_context_set_spell_checking_languages): Deleted.
(webkit_web_context_set_preferred_languages): Deleted.
(webkit_web_context_set_tls_errors_policy): Deleted.
(webkit_web_context_get_tls_errors_policy): Deleted.
(webkit_web_context_set_web_extensions_directory): Deleted.
(webkit_web_context_set_web_extensions_initialization_user_data): Deleted.
(webkit_web_context_set_disk_cache_directory): Deleted.
(webkit_web_context_prefetch_dns): Deleted.
(webkit_web_context_allow_tls_certificate_for_host): Deleted.
(webkit_web_context_set_process_model): Deleted.
(webkit_web_context_get_process_model): Deleted.
(webkit_web_context_set_web_process_count_limit): Deleted.
(webkit_web_context_get_web_process_count_limit): Deleted.
(addOriginToMap): Deleted.
(webkit_web_context_initialize_notification_permissions): Deleted.
(webkit_web_context_send_message_to_all_extensions): Deleted.
(webkit_web_context_set_use_system_appearance_for_scrollbars): Deleted.
(webkit_web_context_get_use_system_appearance_for_scrollbars): Deleted.
(webkit_web_context_get_time_zone_override): Deleted.
(webkitWebContextInitializeNotificationPermissions): Deleted.
(webkitWebContextDownloadStarted): Deleted.
(webkitWebContextInitializeWebExtensions): Deleted.
(webkitWebContextGetProcessPool): Deleted.
(webkitWebContextCreatePageForWebView): Deleted.
(webkitWebContextWebViewDestroyed): Deleted.
(webkitWebContextGetWebViewForPage): Deleted.
* Source/WebKit/UIProcess/API/glib/WebKitWebContext.h.in:
* Source/WebKit/UIProcess/API/glib/WebKitWebContextPrivate.h:
* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
* Source/WebKit/UIProcess/API/glib/webkit.h.in:
* Source/WebKit/UIProcess/API/gtk/WebKitVersion.h.in:
* Source/WebKit/UIProcess/API/wpe/WebKitVersion.h.in:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitInjectedBundleMain.cpp:
(WKBundleInitialize):
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebExtension.cpp:
(webkitWebProcessExtensionCreate):
(webkitWebProcessExtensionDidReceiveUserMessage):
(webkitWebProcessExtensionSetGarbageCollectOnPageDestroy):
(webkitWebExtensionCreate): Deleted.
(webkitWebExtensionDidReceiveUserMessage): Deleted.
(webkitWebExtensionSetGarbageCollectOnPageDestroy): Deleted.
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebExtension.h.in:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebProcessExtension.cpp: Copied from Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebExtension.cpp.
(webkitWebProcessExtensionCreate):
(webkitWebProcessExtensionDidReceiveUserMessage):
(webkitWebProcessExtensionSetGarbageCollectOnPageDestroy):
(webkit_web_process_extension_get_page):
(webkit_web_process_extension_send_message_to_context):
(webkit_web_process_extension_send_message_to_context_finish):
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebProcessExtension.h.in: Added.
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebProcessExtensionPrivate.h: Renamed from Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebExtensionPrivate.h.
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebProcessExtensionManager.cpp: Renamed from Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitExtensionManager.cpp.
(WebKit::WebProcessExtensionManager::singleton):
(WebKit::WebProcessExtensionManager::scanModules):
(WebKit::parseUserData):
(WebKit::WebProcessExtensionManager::initializeWebProcessExtension):
(WebKit::WebProcessExtensionManager::initialize):
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebProcessExtensionManager.h: Renamed from Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitExtensionManager.h.
(WebKit::WebProcessExtensionManager::extension const):
* Source/WebKit/WebProcess/InjectedBundle/API/glib/webkit-web-process-extension.h.in: Renamed from Source/WebKit/WebProcess/InjectedBundle/API/glib/webkit-web-extension.h.in.
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/WebKitWebProcessEnumTypes.cpp.in:
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/WebKitWebProcessEnumTypesGtk3.h.in:
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/WebKitWebProcessEnumTypesGtk4.h.in:
* Source/WebKit/WebProcess/InjectedBundle/API/wpe/WebKitWebProcessEnumTypes.cpp.in:
* Source/WebKit/WebProcess/InjectedBundle/API/wpe/WebKitWebProcessEnumTypesWPE1.h.in: Copied from Source/WebKit/WebProcess/InjectedBundle/API/wpe/WebKitWebProcessEnumTypes.h.in.
* Source/WebKit/WebProcess/InjectedBundle/API/wpe/WebKitWebProcessEnumTypesWPE2.h.in: Renamed from Source/WebKit/WebProcess/InjectedBundle/API/wpe/WebKitWebProcessEnumTypes.h.in.
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
* Source/WebKit/WebProcess/WebPage/glib/WebPageGLib.cpp:
(WebKit::WebPage::sendMessageToWebProcessExtensionWithReply):
(WebKit::WebPage::sendMessageToWebProcessExtension):
(WebKit::WebPage::sendMessageToWebExtensionWithReply): Deleted.
(WebKit::WebPage::sendMessageToWebExtension): Deleted.
* Source/WebKit/WebProcess/WebProcess.h:
* Source/WebKit/WebProcess/WebProcess.messages.in:
* Source/WebKit/WebProcess/glib/WebProcessGLib.cpp:
(WebKit::WebProcess::sendMessageToWebProcessExtension):
(WebKit::WebProcess::sendMessageToWebExtension): Deleted.
* Source/WebKit/gtk/gtk3-urlmap.js.in: Copied from Source/WebKit/gtk/urlmap.js.in.
* Source/WebKit/gtk/gtk3-webkitgtk.toml.in: Copied from Source/WebKit/gtk/webkitgtk.toml.in.
* Source/WebKit/gtk/gtk4-urlmap.js.in: Renamed from Source/WebKit/gtk/urlmap.js.in.
* Source/WebKit/gtk/gtk4-webkitgtk.toml.in: Renamed from Source/WebKit/gtk/webkitgtk.toml.in.
* Source/WebKit/gtk/migrating-to-webkitgtk-6.0.md:
* Source/WebKit/gtk/webkitgtk-web-process-extension.pc.in: Renamed from Source/WebKit/gtk/webkitgtk-web-extension.pc.in.
* Source/WebKit/gtk/webkitgtk-web-process-extension.toml.in: Renamed from Source/WebKit/gtk/webkitgtk-webextension.toml.in.
* Source/WebKit/webkitglib-symbols.map:
* Source/WebKit/wpe/wpe-web-process-extension-uninstalled.pc.in: Renamed from Source/WebKit/wpe/wpe-web-extension-uninstalled.pc.in.
* Source/WebKit/wpe/wpe-web-process-extension.pc.in: Renamed from Source/WebKit/wpe/wpe-web-extension.pc.in.
* Source/WebKit/wpe/wpe-web-process-extension.toml.in: Renamed from Source/WebKit/wpe/wpewebextension.toml.in.
* Source/WebKit/wpe/wpe1-urlmap.js: Copied from Source/WebKit/wpe/urlmap.js.
(const.baseURLApiLevelSuffix):
* Source/WebKit/wpe/wpe2-urlmap.js: Renamed from Source/WebKit/wpe/urlmap.js.
(const.baseURLApiLevelSuffix):
* Source/WebKit/wpe/wpewebkit.toml.in:
* Source/cmake/OptionsGTK.cmake:
* Source/cmake/OptionsWPE.cmake:
* Tools/PlatformWPE.cmake:
* Tools/Scripts/webkitpy/style/checker.py:
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestLoaderClient.cpp:
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestMultiprocess.cpp:
(testProcessPerWebView):
(testMultiprocessWebViewCreateReadyClose):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebProcessExtensions.cpp: Renamed from Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebExtensions.cpp.
(checkTitle):
(testWebProcessExtensionGetTitle):
(inputElementIsUserEdited):
(testWebProcessExtensionInputElementIsUserEdited):
(documentLoadedCallback):
(testDocumentLoadedSignal):
(webProcessTerminatedCallback):
(testWebKitWebViewProcessCrashed):
(scriptDialogCallback):
(runJavaScriptInIsolatedWorldFinishedCallback):
(testWebProcessExtensionIsolatedWorld):
(didAssociateFormControlsCallback):
(testWebProcessExtensionFormControlsAssociated):
(FormSubmissionTest::FormSubmissionTest):
(FormSubmissionTest::~FormSubmissionTest):
(FormSubmissionTest::testFormSubmissionResult):
(FormSubmissionTest::willSendDOMEventCallback):
(FormSubmissionTest::willCompleteCallback):
(FormSubmissionTest::runJavaScriptAndWaitUntilFormSubmitted):
(testWebProcessExtensionFormSubmissionSteps):
(webViewPageIDChanged):
(testWebProcessExtensionPageID):
(UserMessageTest::webViewUserMessageReceivedCallback):
(UserMessageTest::webContextUserMessageReceivedCallback):
(UserMessageTest::UserMessageTest):
(UserMessageTest::~UserMessageTest):
(UserMessageTest::sendMessage):
(UserMessageTest::sendMessageToAllExtensions):
(UserMessageTest::viewUserMessageReceived):
(UserMessageTest::contextUserMessageReceived):
(UserMessageTest::waitUntilViewMessagesReceived):
(UserMessageTest::waitUntilViewMessageReceived):
(UserMessageTest::waitUntilContextMessageReceived):
(readFileDescriptor):
(testWebProcessExtensionUserMessages):
(testWebProcessExtensionWindowObjectCleared):
(beforeAll):
(afterAll):
* Tools/TestWebKitAPI/Tests/WebKitGLib/WebProcessExtensionTest.cpp: Renamed from Tools/TestWebKitAPI/Tests/WebKitGLib/WebExtensionTest.cpp.
(DelayedSignal::DelayedSignal):
(emitDocumentLoaded):
(documentLoadedCallback):
(emitURIChanged):
(uriChangedCallback):
(sendRequestCallback):
(serializeContextMenu):
(serializeNode):
(contextMenuCallback):
(consoleMessageSentCallback):
(emitFormControlsAssociated):
(formControlsAssociatedForFrameCallback):
(formControlsAssociatedCallback):
(emitFormSubmissionEvent):
(willSubmitFormDeprecatedCallback):
(handleFormSubmissionCallback):
(willSendSubmitEventCallback):
(willSubmitFormCallback):
(pageMessageReceivedCallback):
(emitPageCreated):
(pageCreatedCallback):
(extensionMessageReceivedCallback):
* Tools/TestWebKitAPI/Tests/WebKitGLib/WebProcessTest.cpp:
(windowObjectClearedCallback):
(webkit_web_extension_initialize):
* Tools/TestWebKitAPI/Tests/WebKitGLib/WebProcessTest.h:
* Tools/TestWebKitAPI/Tests/WebKitGtk/AutocleanupsTest.cpp:
* Tools/TestWebKitAPI/Tests/WebKitGtk/DOMClientRectTest.cpp:
* Tools/TestWebKitAPI/Tests/WebKitGtk/DOMDOMWindowTest.cpp:
(WebKitDOMDOMWindowTest::testSignals):
(WebKitDOMDOMWindowTest::testDispatchEvent):
(WebKitDOMDOMWindowTest::testGetComputedStyle):
(WebKitDOMDOMWindowTest::runTest):
* Tools/TestWebKitAPI/Tests/WebKitGtk/DOMNodeFilterTest.cpp:
* Tools/TestWebKitAPI/Tests/WebKitGtk/DOMNodeTest.cpp:
* Tools/TestWebKitAPI/Tests/WebKitGtk/DOMXPathNSResolverTest.cpp:
* Tools/TestWebKitAPI/Tests/WebKitGtk/TestDOMDOMWindow.cpp:
(beforeAll):
* Tools/TestWebKitAPI/glib/CMakeLists.txt:
* Tools/TestWebKitAPI/glib/PlatformGTK.cmake:
* Tools/TestWebKitAPI/glib/TestExpectations.json:
* Tools/TestWebKitAPI/glib/WebKitGLib/TestMain.cpp:
(dbusServerConnection):
* Tools/TestWebKitAPI/glib/WebKitGLib/TestMain.h:
(Test::initializeWebProcessExtensionsCallback):
(Test::Test):
(Test::initializeWebProcessExtensions):
(Test::initializeWebExtensionsCallback): Deleted.
(Test::initializeWebExtensions): Deleted.
* Tools/TestWebKitAPI/glib/WebKitGLib/WebViewTest.cpp:
(WebViewTest::extensionProxy):
* Tools/TestWebKitAPI/glib/WebKitGLib/WebViewTest.h:

Canonical link: https://commits.webkit.org/261349@main
